network-config-analyzer/IstioPolicyYamlParser.py

```diff
@@ -167,9 +204,9 @@ def parse_condition(self, condition):
         :return: PeerSet or ConnectionSet (depends on the key) with allowed values
         """
         allowed_elements = {'key': [1, str], 'values': [0, list], 'notValues': [0, list]}
-        allowed_key_values = {'key': ['source.ip', 'source.namespace', 'destination.port']}
+        allowed_key_values = {'key': ['source.ip', 'source.namespace', 'source.principal', 'destination.port']}
         # https://istio.io/latest/docs/reference/config/security/conditions/
-        # TODO: support additional key values: request.headers, remote.ip, source.principal, request.auth.principal,
+        # TODO: support additional key values: request.headers, remote.ip, request.auth.principal,
         #  request.auth.audiences, request.auth.presenter, request.auth.claims, destination.ip, connection.sni
         self.check_fields_validity(condition, 'authorization policy condition', allowed_elements, allowed_key_values)
         for key_elem in allowed_elements.keys():
```

Comment:
```suggestion
        for key_elem in allowed_elements:
```
Iterating the dictionary rather than its keys() is considered more efficient

Reply:
done